src/constbuffer_array.c

```diff
@@ -108,6 +108,114 @@ CONSTBUFFER_ARRAY_HANDLE constbuffer_array_create_empty(void)
     return result;
 }
 
+CONSTBUFFER_ARRAY_HANDLE constbuffer_array_create_from_array_array(const CONSTBUFFER_ARRAY_HANDLE* buffer_arrays, uint32_t buffer_array_count)
+{
+    CONSTBUFFER_ARRAY_HANDLE result;
+
+    if (
+        /*Codes_SRS_CONSTBUFFER_ARRAY_42_009: [ If buffer_arrays is NULL and buffer_array_count is not 0 then constbuffer_array_create_from_array_array shall fail and return NULL. ]*/
+        (buffer_arrays == NULL && buffer_array_count != 0)
+        )
+    {
+        LogError("invalid arguments: const CONSTBUFFER_ARRAY_HANDLE* buffer_arrays=%p, uint32_t buffer_array_count=%" PRIu32, buffer_arrays, buffer_array_count);
+        result = NULL;
+    }
+    else
+    {
+        /*Codes_SRS_CONSTBUFFER_ARRAY_42_001: [ If buffer_arrays is NULL or buffer_array_count is 0 then constbuffer_array_create_from_array_array shall create a new, empty CONSTBUFFER_ARRAY_HANDLE. ]*/
+        if (buffer_arrays == NULL || buffer_array_count == 0)
+        {
+            result = constbuffer_array_create_empty();
+        }
+        else
+        {
+            uint32_t total_buffer_count = 0;
+            uint32_t i;
+            for (i = 0; i < buffer_array_count; ++i)
+            {
+                if (buffer_arrays[i] == NULL)
+                {
+                    /*Codes_SRS_CONSTBUFFER_ARRAY_42_002: [ If any const buffer array in buffer_arrays is NULL then constbuffer_array_create_from_array_array shall fail and return NULL. ]*/
+                    LogError("Invalid arguments: NULL buffer array %" PRIu32, i);
+                    break;
+                }
+                else
+                {
+                    // Overflow check
+                    total_buffer_count += buffer_arrays[i]->nBuffers;
+                    if (total_buffer_count < buffer_arrays[i]->nBuffers)
+                    {
+                        LogError("Array size overflow while checking index %" PRIu32, i);
+                        break;
+                    }
+                }
+            }
+
+            if (i != buffer_array_count)
```

Comment:
>!= [](start = 18, length = 2)

Can be <

Reply:
good point, fixed here and below

---
In reply to: [245158718](https://github.com/Azure/azure-c-shared-utility/pull/248#discussion_r245158718) [](ancestors = 245158718)